ts/torch_handler/trt_llm_handler.py

```diff
@@ -66,55 +68,41 @@ async def preprocess(self, requests):
         data = req_data.get("data") or req_data.get("body")
         if isinstance(data, (bytes, bytearray)):
             data = data.decode("utf-8")
-
-        prompt = data.get("prompt")
-        input_ids = self.tokenizer.encode(
-            prompt, add_special_tokens=True, truncation=True
-        )
-        del data["prompt"]
-        batch_input_ids = torch.tensor([input_ids], dtype=torch.int32)
-        data.update({"batch_input_ids": batch_input_ids})
-
         return data
 
     async def inference(self, data, context):
         generate_kwargs = {
             "end_id": self.tokenizer.eos_token_id,
             "pad_id": self.tokenizer.pad_token_id,
-            "output_sequence_lengths": True,
-            "return_dict": True,
         }
+        prompt = data.get("prompt")
+        streaming = data.get("streaming", False)
+        del data["prompt"]
+        if "streaming" in data:
+            del data["streaming"]
         generate_kwargs.update(data)
+        sampling_params = SamplingParams(**generate_kwargs)
 
-        with torch.no_grad():
-            outputs = self.trt_llm_engine.generate(**generate_kwargs)
-        return outputs, data
-
-    async def postprocess(self, inference_outputs, data, context):
-        if not data.get("streaming", False):
-            output_ids = inference_outputs["output_ids"]
-            sequence_lengths = inference_outputs["sequence_lengths"]
-
-            output_end = sequence_lengths[0]
-            outputs = output_ids[0][0].tolist()
-            output_text = self.tokenizer.decode(outputs)
-            return [output_text]
-
-        # Streaming output
-        for inference_output in inference_outputs:
-            output_ids = inference_output["output_ids"]
-            sequence_lengths = inference_output["sequence_lengths"]
-
-            batch_size, _, _ = output_ids.size()
-            for batch_idx in range(batch_size):
-                output_end = sequence_lengths[batch_idx][0]
-                outputs = output_ids[batch_idx][0][output_end - 1 : output_end].tolist()
-                output_text = self.tokenizer.decode(outputs)
+        outputs = self.trt_llm_engine.generate_async(
+            prompt, streaming=streaming, sampling_params=sampling_params
+        )
+        return outputs, streaming, data
+
+    async def postprocess(self, outputs, streaming, data, context):
+        for output in outputs:
```

Comment:
If we do not await anywhere this coroutine will block execution and we end up only working on a single request. Was this tested with multiple request simultaneously?

Reply:
addressed it..verified it working with 2 concurrent requests